server/controllers/request.js

```diff
@@ -0,0 +1,79 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler")
+
+// @route POST /requests/create
+// @desc Create new request
+// @access Public
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.create(req.body);
+
+  if (request) {
+    res.status(200).json({
+      success: {
+        request: {
+          ...request
+        }
+      }
+    })
+  } else {
+    res.status(400);
+    throw new Error("Invalid request data")
+  }
+});
+
+// @route PUT /request/edit
+// @desc edit request
+// @access Public
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  const request = await Request.findOne({ _id: req.request._id })
+  
+  if (!request) {
+    res.status(404);
+    throw new Error("Request doesn't exist");
+  }
+  request.set(req.body);
+  const updatedRequest = await request.save();
+  res.status(200).json({
+    success: {
+      request: updatedRequest
+    },
+  });
+});
+
+// @route GET /requests
+// @desc get requests of logged in user
+// @access Private
+exports.getUserRequests = asyncHandler(async (req, res, next) => {
+  const requests = await Request.where('requester', req.userId);
```

Comment:
I don't think it's enough to return only the requests where the user is the owner/requester. What if we want to get the requests where the authenticated user is the sitter? That would be usefult for "manage bookings" page for the sitters for example. I think we can either return both, or better, implement [query params](http://expressjs.com/en/api.html#req.query) to let the client pick.

Also, there is a small typo below:
```suggestion
  const requests = await Request.where('requester', req.user.id);
```

Reply:
Added a query param for user type.